projects/plugins/jetpack/_inc/lib/class-jetpack-coauthor-helper.php

```diff
@@ -0,0 +1,210 @@
+
+		/*
+		 * This may need to be updated
+		 * to take into account the different ways we can make requests
+		 * (from a WordPress.com site, from a Jetpack site).
+		 */
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error(
+				'rest_forbidden',
+				__( 'Sorry, you are not allowed to access CoAuthor help on this site.', 'jetpack' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the WPCOM or self-hosted site ID.
+	 *
+	 * @return mixed
+	 */
+	public static function get_site_id() {
+		$is_wpcom = ( defined( 'IS_WPCOM' ) && IS_WPCOM );
+		$site_id  = $is_wpcom ? get_current_blog_id() : Jetpack_Options::get_option( 'id' );
+		if ( ! $site_id ) {
+			return new WP_Error(
+				'unavailable_site_id',
+				__( 'Sorry, something is wrong with your Jetpack connection.', 'jetpack' ),
+				403
+			);
+		}
+		return (int) $site_id;
+	}
+
+	/**
+	 * Get the name of the transient for image generation. Unique per prompt and allows for reuse of results for the same prompt across entire WPCOM.
+	 * I expext "puppy" to always be from cache.
+	 *
+	 * @param  string $prompt - Supplied prompt.
+	 */
+	public static function transient_name_for_image_generation( $prompt ) {
+		return 'jetpack_openai_image_' . md5( $prompt );
+	}
+
+	/**
+	 * Get the name of the transient for text completion. Unique per user, but not per text. Serves more as a cooldown.
+	 */
+	public static function transient_name_for_completion() {
+		return 'jetpack_openai_completion_' . get_current_user_id(); // Cache for each user, so that other users dont get weird cached version from somebody else.
+	}
+
+	/**
+	 * Get text back from WordPress.com based off a starting text.
+	 *
+	 * @param  string $content The content that's already been typed in the block.
+	 * @return mixed
+	 */
+	public static function get_gpt_completion( $content ) {
+		$content = wp_strip_all_tags( $content );
+		$cache   = get_transient( self::transient_name_for_completion() );
+		if ( $cache ) {
+			return array( 'prompts' => array( $cache ) );
+		}
+
+		if ( ( new Status() )->is_offline_mode() ) {
+			return new WP_Error(
+				'dev_mode',
+				__( 'CoAuthor is not available in offline mode.', 'jetpack' )
+			);
+		}
+
+		$site_id = self::get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return $site_id;
+		}
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			if ( ! class_exists( 'OpenAI' ) ) {
+				\require_lib( 'openai' );
+			}
+
+			$result = ( new OpenAI( 'openai' ) )->request_gpt_completion( $content );
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+			// In case of Jetpack we are setting a transient on the WPCOM and not the remote site. I think the 'get_current_user_id' may default for the connection owner at this point but we'll deal with this later.
```

Comment:
I'm not sure I understand this, but like you said, this can be addressed later :)